### PR TITLE
Weather greg

### PR DESCRIPTION
**Pull request by GregoryWilusz**
Please, have a mercy :-)

### CODE REVIEW

File: 3-12-2018/grzesiek/weather/Gemfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+source 'https://rubygems.org'
+
+gem 'sinatra'
+gem 'rspec'
```
Comment (ppietrasik) [inline 3-12-2018/grzesiek/weather/Gemfile]: Gems could be grouped for test and development

File: 3-12-2018/grzesiek/weather/app.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require 'sinatra'
+require './lib/weather_fetcher'
+require 'byebug'
+
+get '/' do
+  begin
+    if params[:location]
+      sanitized_location = WeatherFetcher.sanitize_location(params[:location])
```
Comment (ppietrasik) [inline 3-12-2018/grzesiek/weather/app.rb]: Could be sanitised inside of get_weather_data

File: 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require 'http'
+require 'byebug'
+require 'csv'
+
+module WeatherFetcher
+  module_function
+
+  API_KEY = '5dELbO869g0TMpA6mBrQdQ8zZstnKRPh'
```
Comment (ppietrasik) [inline 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb]: API_KEY could be moved to ENV

File: 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  def sanitize_location(location_to_sanitize)
+    location_to_sanitize.empty? && raise(ArgumentError)
+
+    sanitized_string = location_to_sanitize.tr("ąęźżóńł ", "aezzonl-")
+  end
+
+  def get_location_key(location)
+    response = HTTP.get("http://dataservice.accuweather.com/locations/v1/cities/search?apikey=#{API_KEY}&q=#{location}&details=true")
```
Comment (ppietrasik) [inline 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb]: 'http://dataservice.accuweather.com/' could be separate HOST constant

File: 3-12-2018/grzesiek/weather/spec/app_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      }
+  end
+
+  it 'should allow accessing the home page' do
+    get '/'
+    expect(last_response).to be_ok
+  end
+
+  it 'should receive location key from server' do
+    allow(WeatherFetcher).to receive(:get_location_key).and_return('274663')
```
Comment (ppietrasik) [inline 3-12-2018/grzesiek/weather/spec/app_spec.rb]: Could be move into separate file

File: 3-12-2018/grzesiek/weather/Gemfile (hunk 1 above)
Comment (GregoryWilusz) [reply 3-12-2018/grzesiek/weather/Gemfile]: Thanks for advice, I'll keep it in my mind for the future projects.

File: 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb (hunk 4 above)
Comment (GregoryWilusz) [reply 3-12-2018/grzesiek/weather/lib/weather_fetcher.rb]: Good idea, thanks.

File: 3-12-2018/grzesiek/weather/spec/app_spec.rb (hunk 5 above)
Comment (GregoryWilusz) [reply 3-12-2018/grzesiek/weather/spec/app_spec.rb]: That's definitely right.

File: 3-12-2018/grzesiek/weather/app.rb (hunk 2 above)
Comment (GregoryWilusz) [reply 3-12-2018/grzesiek/weather/app.rb]: You're right. I made corrections and now it's like you suggested.

**Pull request merged**